tests/backends/data/test_ldp.py

```diff
@@ -0,0 +1,770 @@
+    expected = [details_responses[1], details_responses[2]]
+    result = backend.list(target, details=True, new=True)
+    assert isinstance(result, Iterable)
+    assert sorted(result, key=itemgetter("archiveId")) == expected
+
+    # Add archive_2 to history
+    backend.history.append(
+        {
+            "backend": "ldp",
+            "action": "read",
+            "id": "archive_2",
+            "filename": "2020-06-18.gz",
+            "size": 23424233,
+            "timestamp": "2020-10-07T16:37:25.887664+00:00",
+        }
+    )
+
+    # We expect one request to retrieve details for archive 3.
+    get_details_response = (response for response in details_responses[2:])
+
+    # Given a history containing two matching archives and `new` and `details` set to
+    # `True`, the `list` method should yield all archives with additional details,
+    # except the matching ones.
+    expected = [details_responses[2]]
+    result = backend.list(target, details=True, new=True)
+    assert isinstance(result, Iterable)
+    assert sorted(result, key=itemgetter("archiveId")) == expected
+
+    # Add archive_3 to history
+    backend.history.append(
+        {
+            "backend": "ldp",
+            "action": "read",
+            "id": "archive_3",
+            "filename": "2020-06-19.gz",
+            "size": 23424233,
+            "timestamp": "2020-10-07T16:37:25.887664+00:00",
+        }
+    )
+
+    # Given a history containing all matching archives and `new` and `details` set to
+    # `True`, the `list` method should yield nothing.
+    expected = []
+    result = backend.list(target, details=True, new=True)
+    assert isinstance(result, Iterable)
+    assert list(result) == expected
+
+
+def test_backends_data_ldp_data_backend_read_method_without_raw_ouput(
+    ldp_backend, caplog, monkeypatch
+):
+    """Test the `LDPDataBackend.read method, given `raw_output` set to `False`, should
+    log a warning message.
+    """
+
+    class MockResponse:
+        """Mock the requests response."""
+
+        def __enter__(self):
+            return self
+
+        def __exit__(self, *args):
+            pass
+
+        def raise_for_status(self):
+            """Ignored."""
+
+        def iter_content(self, chunk_size):
+            """Fake content file iteration."""
+            # pylint: disable=no-self-use,unused-argument
+            yield
+
+    def mock_requests_get(url, stream=True, timeout=None):
+        """Mock the request get method."""
+        # pylint: disable=unused-argument
+        return MockResponse()
+
+    def mock_get(url):
+        """Mock the OVH client get request."""
+        # pylint: disable=unused-argument
+        return {"filename": "archive_name", "size": 10}
+
+    backend = ldp_backend()
+
+    monkeypatch.setattr(requests, "get", mock_requests_get)
+    backend = ldp_backend()
+    monkeypatch.setattr(backend, "_url", lambda *_: "/")
+    monkeypatch.setattr(backend.client, "get", mock_get)
```

Comment:
Could `requests_mock` fixture fit to mock the `requests` response in this testcase? :point_right: [requests_mock from Pytest](https://requests-mock.readthedocs.io/en/latest/pytest.html)

Reply:
That's a great idea) This would indeed simplify the current tests that use the MockResponse instead.
However, I'm hesitating about adding a new dependency to `setup.cfg` in this branch as this file frequently changes (to avoid merge conflicts). Could this be done after or at the end of the v4 release? WDYT?

Reply:
That's not a problem for me. We can add it as an improvement in an issue to not forget to do it